core/src/Streamly/Internal/Data/Fold.hs

```diff
@@ -434,6 +435,50 @@ scan (Fold stepL initialL extractL) (Fold stepR initialR extractR) =
 
     extract = extractR . snd
 
+-- | Scan the output of first 'Fold' to change it using second 'Fold'.
+-- If second fold terminates it will restart again from initial state.
+-- /Pre-release/
+scanMany :: Monad m => Fold m a b -> Fold m b c -> Fold m a c
+scanMany (Fold stepL initialL extractL) (Fold stepR initialR extractR) =
+    Fold step initial extract
+
+    where
+
+    {-# INLINE runStep #-}
+    runStep actionL sR nxt = do
+        rL <- actionL
+        case rL of
+            Done bL -> do
+                rR <- stepR sR bL
+                case rR of
+                    Partial sR1 -> Done <$> extractR sR1
+                    Done bR -> return $ Done bR
+            Partial sL -> do
+                !b <- extractL sL
+                rR <- stepR sR b
+                case rR of
+                    Partial sR1 -> return $ Partial (sL, sR1, nxt)
+                    Done b1 -> do
+                        if isNothing nxt
+                        then return $ Partial (sL, sR, Just b1)
+                        else do
+                            r <- initialR
+                            case r of
+                                Partial sR1 -> runStep actionL sR1 Nothing
+                                Done b2 -> return $ Done b2
```

Comment:
This is not correct. We need to scan the input of the right fold using the left fold. So in `Done bL` case we need to initialize the left fold again.